pandas/core/groupby/generic.py

```diff
@@ -321,13 +321,12 @@ def _aggregate_multiple_funcs(self, arg):
                 obj = copy.copy(obj)
                 obj._reset_cache()
                 obj._selection = name
-            results[name] = obj.aggregate(func)
+            results[base.OutputKey(label=name, position=idx)] = obj.aggregate(func)
 
         if any(isinstance(x, DataFrame) for x in results.values()):
             # let higher level handle
-            return results
-
-        return DataFrame(results, columns=columns)
+            return {key.label: value for key, value in results.items()}
+        return DataFrame(self._wrap_aggregated_output(results), columns=columns)
```

Comment:
Is the DataFrame constructor still required here?

Reply:
In the test
```
pytest pandas/tests/groupby/aggregate/test_aggregate.py::test_aggregate_item_by_item
```
when we get here we have
```
(Pdb) results
{OutputKey(label='<lambda>', position=0): A
bar    3
foo    5
Name: B, dtype: int64}
(Pdb) self._wrap_aggregated_output(results)
A
bar    3
foo    5
Name: <lambda>, dtype: int64
(Pdb) type(self._wrap_aggregated_output(results))
<class 'pandas.core.series.Series'>
```

Reply:
@WillAyd have updated with a call to `.to_frame` (if necessary)